hack/gen-tls-certs.sh

```diff
@@ -0,0 +1,179 @@
+        },
+        "profiles": {
+            "internal": {
+                "expiry": "${CA_EXPIRATION}",
+                "usages": [
+                    "signing",
+                    "key encipherment",
+                    "server auth",
+                    "client auth"
+                ]
+            },
+            "server": {
+                "expiry": "${CA_EXPIRATION}",
+                "usages": [
+                    "signing",
+                    "key encipherment",
+                    "server auth"
+                ]
+            },
+            "client": {
+                "expiry": "${CA_EXPIRATION}",
+                "usages": [
+                    "signing",
+                    "key encipherment",
+                    "client auth"
+                ]
+            }
+        }
+    }
+}
+EOF
+
+    cat <<EOF >ca-csr.json
+{
+    "CN": "${COMMON_NAME}",
+    "key": {
+        "algo": "rsa",
+        "size": 2048
+    },
+    "names": [
+        {
+            "C": "${COUNTRY}",
+            "L": "${LOCALITY}",
+            "O": "${ORGANIZATION}",
+            "ST": "${STATE_OR_PROVINCE_NAME}",
+            "OU": "${ORGANIZATION_UNIT}"
+        }
+    ]
+}
+EOF
+
+    $CFSSL_BIN gencert -initca ca-csr.json | $CFSSLJSON_BIN -bare ca -
+    popd
+}
+
+function gen_component_certs() {
+    local KIND=$1
+    local COMP=$2
+    local COMP_DIR=$TLS_DIR/$KIND/$2
+    local PROFILE=$3
+    echo "generating $COMP server certs"
+    mkdir -p $COMP_DIR
+    pushd $COMP_DIR
+    $CFSSL_BIN print-defaults csr \
+        | jq ".hosts=[\"127.0.0.1\",\"::1\",\"${TIDB_CLUSTER}-${COMP}\",\"${TIDB_CLUSTER}-${COMP}.${NAMESPACE}\",\"${TIDB_CLUSTER}-${COMP}.${NAMESPACE}.svc\",\"${TIDB_CLUSTER}-${COMP}-peer\",\"${TIDB_CLUSTER}-${COMP}-peer.${NAMESPACE}\",\"${TIDB_CLUSTER}-${COMP}-peer.${NAMESPACE}.svc\",\"*.${TIDB_CLUSTER}-${COMP}-peer\",\"*.${TIDB_CLUSTER}-${COMP}-peer.${NAMESPACE}\",\"*.${TIDB_CLUSTER}-${COMP}-peer.${NAMESPACE}.svc\"]" \
+        | jq ".CN=\"${COMMON_NAME}\"" \
+        > ${COMP}-server.json
+    $CFSSL_BIN gencert -ca=$CA_DIR/ca.pem -ca-key=$CA_DIR/ca-key.pem -config=$CA_DIR/ca-config.json -profile=$PROFILE ${COMP}-server.json | $CFSSLJSON_BIN -bare ${COMP}-server
+    popd
+}
+
+function create_cluster_secret() {
+    local COMP=$1
+    $KUBECTL_BIN get secret ${TIDB_CLUSTER}-${COMP}-cluster-secret --namespace=${NAMESPACE} && \
+        $KUBECTL_BIN delete secret ${TIDB_CLUSTER}-${COMP}-cluster-secret --namespace=${NAMESPACE}
+    $KUBECTL_BIN create secret generic ${TIDB_CLUSTER}-${COMP}-cluster-secret --namespace=${NAMESPACE} \
+        --from-file=tls.crt=${TLS_DIR}/cluster/${COMP}/${COMP}-server.pem \
+        --from-file=tls.key=${TLS_DIR}/cluster/${COMP}/${COMP}-server-key.pem \
+        --from-file=ca.crt=${CA_DIR}/ca.pem
+}
+
+hack::ensure_cfssl
+
+echo "TLS_DIR: $TLS_DIR"
+echo "CA_DIR: $CA_DIR"
+
+if [[ -d $TLS_DIR ]]; then
+    echo "tls dir exists, removing"
+    rm -rf $TLS_DIR
+fi
+
```

Comment:
Add a `usage` function to explain the usage as we do in `e2e.sh`?

Reply:
done